llama_stack/distribution/routers/routing_tables.py

```diff
@@ -500,6 +500,8 @@ async def register_tool_group(
                 )
             provider_id = list(self.impls_by_provider_id.keys())[0]
 
+        # parse tool group to the type if dict
+        tool_group = parse_obj_as(ToolGroupDef, tool_group)
```

Comment:
nit: `parse_obj_as` is deprecated. `TypeAdapter(ToolGroupDef).validate_python(...)` is the recommended replacement.

Reply:
done.